src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -976,10 +979,21 @@ def post_partial_compile_validation(
         model = pathlib.Path(self.project._test_project_dir, "main.cf").read_text()
         self.project.compile(model)
 
-        # Check that we have as many resource sets as there are services
-        assert get_resource_sets(self.project).keys() == {
-            str(srv.id) for srv in self.services.values() if not srv.deleted and not srv.state == "ordered"
+        # Check each service in the inventory has its own resource set
+        all_resource_sets = get_resource_sets(self.project).keys()
```

Comment:
This doesn't always hold (for systemtenant, we always only emit the resource set of the systemtenant).